site/developer/developer_guide.md

```diff
@@ -0,0 +1,238 @@
+
+The `core` module contains the main source code for the project. This includes Jython modules and Java classes, as well as typedef files, alias definitions, and the message bundle. There are unit tests related to this module.
+
+Alias definitions are discussed in more detail [here](#alias-definitions). 
+
+The `installer` module builds the final installer zip file. It includes the assembly definitions, start scripts for each tool for Linux and Windows platforms, and configurations for variable injection and logging. 
+
+A single installer zip file is built under the `WLSDEPLOY_HOME/installer/target` directory.
+
+There are detailed instructions for building the project [here](#building-weblogic-deploy-tool)
+
+## Functional Areas
+
+This section contains information about the operation of functional areas within the project.
+
+### Creator and Deployer Class Hierarchies
+
+The creation of individual folders and attributes within the `topology` section of the domain model is accomplished using subclasses of the Jython class `Creator`, in the module `wlsdeploy.tool.create.creator.py`.  The `Creator` class provides base methods to recurse through nested folders in the domain model, create or update those folders, and set or update their attributes. Each subclass can override these methods to account for variations in behavior for different functional areas.
+
+For example, the `SecurityProviderCreator` subclass overrides the method `_create_named_subtype_mbeans` with special processing to remove all existing security providers, and re-create them from the data in the model. 
+
+The update of folders and attributes in the `resources` section of the domain model follows a similar pattern, but the base class for these modules is `Deployer` in the module `wlsdeploy.tool.deploy.deployer.py`.
+
+The class `TopologyUpdater` is a special subclass of `Deployer` that is used to update elements in the `topology` section after their initial creation.
+
+## Alias Definitions
+
+WebLogic Deploy Tool uses a set of JSON configuration files to map folders and attributes in the model to the corresponding WLST MBeans and their attributes. These mappings are referred as 'aliases' throughout the project code and documentation. Each element in the alias definition file has detailed properties that assist in this mapping.
+
+The model's folder and attribute names usually match the names of the corresponding elements in WLST. For cases where the names of WLST elements may change across WebLogic Server releases, the names should match the names in the 12.2.1.3 release. The unit test `AttributesTestCase` verifies that this convention is used, and identifies a few exceptions.
+
+Attributes that are introduced after the 12.2.1.3 release should, in most cases, match the name of the first WebLogic Server release in which they appear. The unit test `AttributesTestCase` will ignore these for now, since they will not be present in the 12.2.1.3 alias structure.
```

Comment:
since  -> because (since-don't use to mean for the reason of; use because. Use since only for matters of time, for example, since the last time the program was run.)

Reply:
check, thanks for the tip